### PR TITLE
shogun:lib Replace ASSERT with REQUIRE

### PR DESCRIPTION
**Pull request by somya-anand**
Fix issue #2702 in file src/shogun/lib/SGVector.cpp

### CODE REVIEW

File: src/shogun/lib/SGVector.cpp [hunk 1]
```diff
@@ -304,7 +303,7 @@ void SGVector<T>::add(const SGSparseVector<T>& x)
 		for (int32_t i=0; i < x.num_feat_entries; i++)
 		{
 			index_t idx = x.features[i].feat_index;
-			ASSERT(idx < vlen)
+			REQUIRE(idx < vlen, "Feature index should be less than %d.\n", vlen);
```
Comment (iglesias) [inline src/shogun/lib/SGVector.cpp]: Can you display here idx as well? Maybe with `Features index (%d) should`...

File: src/shogun/lib/SGVector.cpp [hunk 2]
```diff
@@ -371,7 +370,7 @@ template <>
 void SGVector<bool>::display_vector(const bool* vector, int32_t n, const char* name,
 		const char* prefix)
 {
-	ASSERT(n>=0)
+	REQUIRE(n>=0, "Vector size can not be negative.\n");
```
Comment (iglesias) [inline src/shogun/lib/SGVector.cpp]: Here and below maybe it is good to display n as well.

Comment (iglesias) [inline src/shogun/lib/SGVector.cpp]: Nah, never mind about that one. It should be understandable enough to say that a negative value was passed.

**Comment by iglesias** [comment]
All right, looking good. We wait for Travis and then we merge.

File: src/shogun/lib/SGVector.cpp (hunk 2 above)
Comment (karlnapf) [inline src/shogun/lib/SGVector.cpp]: No, please put the idx. Provides help what is going wrong.
https://github.com/shogun-toolbox/shogun/wiki/Assertions

File: src/shogun/lib/SGVector.cpp [hunk 3]
```diff
@@ -868,7 +867,7 @@ void SGVector<complex128_t>::load(CFile* loader)
 
 template<class T> void SGVector<T>::save(CFile* saver)
 {
-	ASSERT(saver)
+	REQUIRE(saver, "Requires a valid 'c FILE pointer'\n");
```
Comment (karlnapf) [inline src/shogun/lib/SGVector.cpp]: "Provided CFile pointer is NULL\n"

File: src/shogun/lib/SGVector.cpp [hunk 4]
```diff
@@ -518,7 +517,7 @@ template <>
 void SGVector<complex128_t>::display_vector(const complex128_t* vector, int32_t n,
 		const char* name, const char* prefix)
 {
-	ASSERT(n>=0)
+	REQUIRE(n>=0, "Vector size can not be negative.\n");
```
Comment (karlnapf) [inline src/shogun/lib/SGVector.cpp]: see above, please display index, and it is spelled "cannot", but it should really be "must not"

**Pull request merged**